### PR TITLE
Add Prim's algorithm to find MST (Complexity : O(v^2))

### PR DESCRIPTION
**Pull request by tanvi15**
Issue #34 
Graph represented as adjacency matrix

### CODE REVIEW

File: Algorithms/Graphs/MST/prim.cpp [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+       for( i=0 ; i<v ; i++){
+            cout<<mst[i]<<" to "<<i<<" weighs "<<graph[i][mst[i]]<<endl;
+
+       }
+
+
+    }
+
+
```
Comment (sahilbansal17) [inline Algorithms/Graphs/MST/prim.cpp]: @tanvi15 
- Add a `return 0;` statement at the end of **main** function.
- Also, please remove the additional spaces at the end.
- Rest all seems perfect. There can be a comment at the end of program elaborating on the complexity of the algorithm.

File: Algorithms/Graphs/MST/prim.cpp [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+int main(){
+    cout<<"Enter number of vertices of the graph"<<endl;
+    int v;
+    cin>>v;
+
+    int graph[v][v];
+    int i,j; //Iterators
```
Comment (arnav-mandal1234) [inline Algorithms/Graphs/MST/prim.cpp]: should be i, j;

File: Algorithms/Graphs/MST/prim.cpp [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    for( i=0 ; i<v ; i++ ){
+        if( !included[i] && keys[i] < mini){
+            mini = keys[i];
+            index = i;
+        }
+    }
+    return index;
+}
+
+
```
Comment (arnav-mandal1234) [inline Algorithms/Graphs/MST/prim.cpp]: redundant newlines

File: Algorithms/Graphs/MST/prim.cpp [hunk 4]
```diff
@@ -0,0 +1,100 @@
+#include <iostream>
+using namespace std;
+
+int findMin( int* keys, bool* included, int v){
+    int mini = 9999;
```
Comment (arnav-mandal1234) [inline Algorithms/Graphs/MST/prim.cpp]: change it to INT_MAX

File: Algorithms/Graphs/MST/prim.cpp [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+       for( j=0 ; j<v ; j++){
+            if ( !included[j] && graph[minKey][j] && graph[minKey][j] < keys[j]){
+                mst[j] = minKey;
+                keys[j] = graph[minKey][j];
+            }
+       }
+
+       cout<<"Minimum spanning tree has been generated."<<endl;
+
+       for( i=0 ; i<v ; i++){
```
Comment (arnav-mandal1234) [inline Algorithms/Graphs/MST/prim.cpp]: loop should be like  for (int i = 0; i < n; i ++)
Indentation problems

File: Algorithms/Graphs/MST/prim.cpp [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+            cout<<mst[i]<<" to "<<i<<" weighs "<<graph[i][mst[i]]<<endl;
```
Comment (arnav-mandal1234) [inline Algorithms/Graphs/MST/prim.cpp]: proper spacing required to increase readability.

File: Algorithms/Graphs/MST/prim.cpp (hunk 2 above)
Comment (sahilbansal17) [reply Algorithms/Graphs/MST/prim.cpp]: @arnav-mandal1234 is referring to the indentation i.e. there should be a space after the comma.

Comment (tanvi15) [reply Algorithms/Graphs/MST/prim.cpp]: Ohh, okay sure

**Comment by tanvi15** [comment]
Changes made.

File: Algorithms/Graphs/MST/prim.cpp (hunk 2 above)
Comment (arnav-mandal1234) [reply Algorithms/Graphs/MST/prim.cpp]: These type of things increase readability.

**Comment by arnav-mandal1234** [comment]
@tanvi15 check the for loops.
I have mentioned it in my review.

File: Algorithms/Graphs/MST/prim.cpp [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+for finding the Minimum Spanning Tree of a graph represented
+as an adjacency matrix (2d matrix).
+
+The time complexity of this algorithm is O(v^2), where
+v is the number of vertices, since adjacent vertices of each
+vertex are being iterated through (v*v).
+
+*/
+
+#include <iostream>
```
Comment (ramitsawhney27) [inline Algorithms/Graphs/MST/prim.cpp]: No need for iostream if you're using bits/stdc++

File: Algorithms/Graphs/MST/prim.cpp [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+
+    cout<<"Enter number of edges of the graph"<<endl;
+    int e;
+    cin>>e;
+
+    cout<<"Enter the vertices and their edge weights in the following format :"<<endl;
+    cout<<"Vertex1 Vertex2 Weight"<<endl;
+    cout<<"PS : First vertex should be 0."<<endl;
+
+    while(e--){
```
Comment (ramitsawhney27) [inline Algorithms/Graphs/MST/prim.cpp]: Never modify the number of edges variable. What if you need it for other functions/purposes later? Either use a for loop or store the value so you always have the number of edges retained.

**Comment by tanvi15** [comment]
@ramitsawhney27 done.

**Comment by sahilbansal17** [comment]
@ramitsawhney27 One more review required.

**Comment by aditya81070** [comment]
@sahilbansal17 I have reviewed this code.

**Comment by tanvi15** [comment]
@sahilbansal17 I have 3 approvals, can it be merged now?

**Pull request merged**